src/main/scala/com/redislabs/provider/redis/rdd/RedisRDD.scala

```diff
@@ -80,8 +98,14 @@ class RedisListRDD(prev: RDD[String],
   def getSET(nodes: Array[RedisNode], keys: Iterator[String]): Iterator[String] = {
     groupKeysByNode(nodes, keys).flatMap { case (node, nodeKeys) =>
       val conn = node.endpoint.connect()
-      val setKeys = filterKeysByType(conn, nodeKeys, "set")
-      val res = setKeys.flatMap(conn.smembers).iterator
+      val res: Iterator[String] = nodeKeys.flatMap{k =>
+        Try(conn.smembers(k).toSet) match {
+          case Failure(e: JedisDataException) if Option(e.getMessage).getOrElse("").contains("WRONGTYPE") => None
```

Comment:
The exception handling seems to be repeated in several places. Can we extract it in a common function?

Reply:
Sure thing

Reply:
Done